packages/@react-spectrum/button/src/ButtonGroup.tsx

```diff
@@ -35,7 +36,9 @@ export function ButtonGroup<T>(props: CollectionBase<T> & SpectrumButtonGroupPro
     ...otherProps
   } = props;
 
-  let state = useButtonGroupState({...props, selectionMode});
+  let {direction} = useLocale();
+  let flipKeyOrder = direction === 'rtl' && orientation === 'horizontal';
```

Comment:
the key array should only be reversed in a horizontal orientation case cuz that is the only one that matters in RTL

debated with myself on whether or not to make this `flipKeyOrder` or `dir = direction === 'rtl' && orientation === 'horizontal' ? 'rtl' : 'ltr'` but `dir` doesn't truly reflect the actual direction so I went with `flipKeyOrder`